lnrpc/routerrpc/router_backend.go

```diff
@@ -29,7 +29,7 @@ type RouterBackend struct {
 	// routes.
 	FindRoutes func(source, target routing.Vertex,
 		amt lnwire.MilliSatoshi, restrictions *routing.RestrictParams,
-		numPaths uint32, finalExpiry ...uint16) (
+		numPaths uint32, finalExpiry ...int32) (
```

Comment:
Any strong reason to move to signed integers here? Virtually none of these fields are permitted to be negative. Historically, we've also used `uint16` for time lock deltas as we typically don't need anything more than a few weeks into the future.

Reply:
Got bitten a few times now with negative numbers after subtraction (deltas can be negative) and want to prevent it from happening again to me or a future dev working with heights.

Reply:
In general, I think it is good to consistently use `int32` for heights and block deltas everywhere throughout the code base.